src/main/kotlin/com/desmonddavid/bots/digitforumbot/bots/NewPostsBot.kt

```diff
@@ -63,13 +63,13 @@ object NewPostsBot : TelegramLongPollingBot() {
                 }
             }
             if (update.message.text.equals("/stop") || update.message.text.equals("/stop@DigitForumBot")) {
-                val filteredList = arrayListOf<Long>()
-                println("Received stop for chat: ${update.message.chatId}")
-                activeChatIds.filterTo(filteredList, { it!=update.message.chatId })
-                activeChatIds = filteredList
-                removeChatIdFromFile(update.message.chatId)
-                println("Current active chat ids: $activeChatIds")
-                sendTextMessage(update.message.chatId, "DigitForumBot has been deactivated for this chat. Goodbye!")
+                if(activeChatIds.contains(update.message.chatId)) {
+                    println("Received stop for chat: ${update.message.chatId}")
+                    activeChatIds.removeIf({ it==update.message.chatId })
```

Comment:
Move this below `removeChatIdFromFile` so that in case writing to file fails then the chatId will remain in the list. Refer to the `/start` condition above.

Reply:
Also, remove the brackets for removeIf() and instead use removeIf {} directly.